datafusion/core/src/catalog_common/mod.rs

```diff
@@ -185,9 +185,8 @@ pub fn resolve_table_references(
                 let _ = s.as_ref().visit(visitor);
             }
             DFStatement::CreateExternalTable(table) => {
-                visitor
-                    .relations
-                    .insert(ObjectName(vec![Ident::from(table.name.as_str())]));
+                let idents: Vec<Ident> = table.name.split('.').map(Ident::from).collect();
```

Comment:
This part is not necessary for the bug fix. Not sure if I should leave it here to maintain consistency of external table name handling.

Reply:
I think the type of [table.name](https://github.com/apache/datafusion/blob/3892499b28135f9862ffac3e4f19524bb5b77c59/datafusion/sql/src/parser.rs#L184) needs to be changed from `String` to `ObjectName`.